### PR TITLE
#170947639: Users should be able to chat on Barefoot Nomad

### PR DESCRIPTION
**Pull request by Kagorora**
#### What does this PR do?

- Allow users to chat on Barefoot Nomad

#### Description of Task to be completed?

- [x]  Design UI

- [x]  Send Request(Message)

- [x]  Receive Message(View messages)

- [x]  Auto-scroll for a new message

#### How should this be manually tested?

- clone the project

      https://github.com/Stackup-Rwanda/knights-bn-frontend.git

- Go to the branch

      git checkout ft-chat-bot-170947639

- Install the packages

      npm install

- Start the server

       npm run dev-start

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [#170947639](https://www.pivotaltracker.com/story/show/170947639)

#### Screenshots (if appropriate)

    CHAT

Screen Shot 2020-06-04 at 19 04 42

#### Questions:

-  N/A

### CODE REVIEW

**Review by EugeneMunya — APPROVED** [review]

**Review by Niyonsengaeric — APPROVED** [review]

**Review by T2Wil — APPROVED** [review]
Nice work @Kagorora

File: .babelrc [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "plugins": [
     "@babel/plugin-proposal-class-properties",
     "@babel/plugin-transform-regenerator",
-    "@babel/plugin-transform-runtime",
-    "transform-remove-console"
+    "@babel/plugin-transform-runtime"
+    // "transform-remove-console"
```
Comment (kagabof) [inline .babelrc]: Why do we have commented code?

File: src/components/Chat.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import $ from 'jquery';
+import { withRouter } from 'react-router';
+import { callApiThunk as thunk } from '../redux/thunk/index';
+import { chats as allChat } from '../redux/actions/index';
+import '../assets/styles/home.scss';
+
+const ioClient = io.connect('https://knights-bn-backnd.herokuapp.com/');
+
+export class Home extends Component {
+  constructor(props) {
```
Comment (kagabof) [inline src/components/Chat.js]: You should be using es6.

File: src/components/Chat.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    const { isHidden } = this.state;
+    this.setState({
+      ...this.state,
+      isHidden: !isHidden,
+    });
+  };
+
+  render() {
+    ioClient.on('send-chat-message-2', (element) => {
+      if (element.from === this.state.author) {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/Chat.js:82]: I think You should mark it as resolved you should consider refactoring.

File: src/components/Chat.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        $('#message-thread').append($('<li class="mines">').html(`<div class="author">${element.from}</div> <div class="message">${element.message}</div> <div class="time">${moment().calendar()}</div>`));
+        $('#message-side').animate({ scrollTop: $('#message-side').prop('scrollHeight') }, 500);
+      } else {
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/Chat.js:84]: I think You should mark it as resolved you should consider refactoring.

File: src/components/Chat.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+              {author.toLowerCase()}
+            </h3>
+          </div>
+          <div className="chart-body">
+            <div className="message-side">
+              <ul className="message-thread">
+                {chats
+                  && chats.map((msg) => {
+                    if (msg.senderName === this.state.author) {
+                      return (
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/Chat.js:108]: I think You should mark it as resolved you should consider refactoring.

File: src/components/Chat.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+                        <li key={msg.id} className="mines">
+                          <div className="author">{msg.senderName}</div>
+                          <div className="message">{msg.message}</div>
+                          <p className="time">
+                            {moment(msg.createdAt).calendar()}
+                          </p>
+                        </li>
+                      );
+                    }
+                    return (
```
[the remark this answers is not in the document]
Comment (kagabof) [reply src/components/Chat.js:118]: I think You should mark it as resolved you should consider refactoring.

**Review by kagabof — CHANGES_REQUESTED** [review]

**Review by kagabof — APPROVED** [review]

**Pull request merged**